src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3574,78 +3574,170 @@ internal Binder GetBinder(CSharpSyntaxNode syntaxNode)
+                            mergePartialMethods(ref membersByName, name, currentMethod, prevMethod, diagnostics);
+                            break;
+
+                        case (SourcePropertySymbol currentProperty, SourcePropertySymbol prevProperty):
+                            mergePartialProperties(ref membersByName, name, currentProperty, prevProperty, diagnostics);
+                            break;
+
+                        case (SourcePropertyAccessorSymbol, SourcePropertyAccessorSymbol):
+                            break; // accessor symbols and their diagnostics are handled by processing the associated property
+
+                        default:
+                            throw ExceptionUtilities.UnexpectedValue((symbol, prev));
+                    }
+                }
+
+                foreach (var symbol in membersBySignature.Values)
+                {
+                    switch (symbol)
+                    {
+                        case SourceOrdinaryMethodSymbol method:
+                            // partial implementations not paired with a definition
+                            if (method.IsPartialImplementation && method.OtherPartOfPartial is null)
                             {
-                                // Avoid mutating the cached dictionary and especially avoid doing this possibly on multiple threads in parallel.
-                                membersByName = new Dictionary<ReadOnlyMemory<char>, ImmutableArray<Symbol>>(membersByName, ReadOnlyMemoryOfCharComparer.Instance);
+                                diagnostics.Add(ErrorCode.ERR_PartialMethodMustHaveLatent, method.GetFirstLocation(), method);
                             }
+                            else if (method is { IsPartialDefinition: true, OtherPartOfPartial: null, HasExplicitAccessModifier: true })
+                            {
+                                diagnostics.Add(ErrorCode.ERR_PartialMethodWithAccessibilityModsMustHaveImplementation, method.GetFirstLocation(), method);
+                            }
+                            break;
 
-                            membersByName[name] = FixPartialMember(membersByName[name], prevPart, methodPart);
-                        }
+                        case SourcePropertySymbol property:
+                            if (property.OtherPartOfPartial is null)
+                            {
+                                diagnostics.Add(
+                                    property.IsPartialDefinition ? ErrorCode.ERR_PartialPropertyMissingImplementation : ErrorCode.ERR_PartialPropertyMissingDefinition,
+                                    property.GetFirstLocation(),
+                                    property);
+                            }
+                            break;
+
+                        case SourcePropertyAccessorSymbol:
+                            break; // handled by SourcePropertySymbol case
```

Comment:
This comment isn't particularly clear. Do you mean a case in SourcePropertySymbol? If not, what are you referring to?

Reply:
I'll think of better wording, what I am meaning to say is that diagnostics for a missing declaration of a partial accessor are handled by `mergePartialProperties`.